packages/static-wado-webserver/lib/controllers/server/byteRangeRequest.mjs

```diff
@@ -17,44 +25,55 @@ export default function byteRangeRequest(options) {
     return fs.existsSync(fullPath);
   }
 
-  async function rangeResponse(req,res,range) {
+  function findExtension(path, accept) {
+    const testExtension = extensions[accept] || ".mht";
+
+    if (!testExtension) return "";
+    const fullPath = `${baseDir}/${path}${testExtension}`;
+    if (fs.existsSync(fullPath) ) {
+      return testExtension;
+    }
+    if( fs.existsSync(fullPath+'.gz') ) {
+      return testExtension+'.gz';
+    }
+    return "";
+  }
+
+  async function rangeResponse(req, res, range, extension) {
     // Better hope the range is a simple - range
-    const bytes = range.substring(6).split('-');
-    const path = `${baseDir}/${req.path}`;
+    const bytes = range.substring(6).split("-");
+    const path = `${baseDir}/${req.path}${extension}`;
     if (!fs.existsSync(path)) {
-      res.status(400).text("Not found");
+      res.status(400).send("Not found");
       return;
     }
     const rawdata = await fs.promises.readFile(path);
     const { length } = rawdata;
     bytes[0] ||= 0;
-    bytes[1] ||= length;
-    bytes[1] = Math.min(bytes[1], length);
-    const data = rawdata.slice(bytes[0], bytes[1]);
-    res.setHeader("content-type", "multipart/related");
-    res.setHeader("Access-Control-Allow-Headers", "Content-Range");
-    res.setHeader("Content-Range", `${bytes[0]}-${bytes[1]}/${length}`);
+    bytes[1] ||= length-1;
```

Comment:
shouldn't this really be the length?

Reply:
Absolutely not - if you set it that way it kills the video client.  The range is inclusive at both ends.